plugins/outputs/http/http.go

```diff
@@ -162,7 +168,17 @@ func (h *HTTP) Write(metrics []telegraf.Metric) error {
 }
 
 func (h *HTTP) write(reqBody []byte) error {
-	req, err := http.NewRequest(h.Method, h.URL, bytes.NewBuffer(reqBody))
+	var reqBodyBuffer io.Reader = bytes.NewBuffer(reqBody)
+	var err error
+
+	if h.ContentEncoding == "gzip" && h.Method == http.MethodPost {
```

Comment:
Why only for POST method?

Reply:
That's what was requested in #4472. It will add a bit of overhead to requests which don't have a body, but I wouldn't normally consider such small optimisations. Maybe @ madisonleavo can offer additional insight.

Reply:
All of the requests currently have a body in this plugin, and it is hard to imagine a new data format that wouldn't produce it, perhaps a null data format?  At any rate we currently do POST and PUT and I think they should both use the encoding if it is requested.

Reply:
Cool! Done in 26ee478.